File: back/src/main/java/br/com/nostramassa/gestao/dtos/pedido/PedidoDTO.java
package br.com.nostramassa.gestao.dtos.pedido;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import br.com.nostramassa.gestao.enums.StatusPedidoEnum;
import br.com.nostramassa.gestao.enums.TipoDePagamentoEnum;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PedidoDTO {

	private Long id;
	
	private Long idPedido;
	
	private LocalDateTime dataPedido;
	
	private boolean entrega;
	
	private boolean apalito;
	
	private TipoDePagamentoEnum tipoPagamento;
	
	private StatusPedidoEnum status;
	
	private Double valor;
	
	private Double taxaEntrega;
	
	private String enderecoDescricao;
	
	private String observacao;
	
	private Long idCliente;
	
	private String telefone;
	
	private String nome;
	
	private String bairro;

	private String rua;

	private String numero;

	private String bloco;

	private String apartamento;

	private String complemento;
	
	private String pedidoRelatorio;
	
	private List<PedidoItemDTO> itensPedido = new ArrayList<>();
	
}

File: front/src/app/shared/models/PedidoDTO.ts
import { PizzaSaborDTO } from "./PizzaDTO";

export class PedidoDTO {
    id: number = 0;
    idPedido?: number = 0;
    dataPedido: string = '';
    entrega : boolean= true;
    apalito : boolean = false;
    tipoPagamento: string = '';
    status?: string = '';
    valor: number = 0;
    taxaEntrega: number = 0;
    enderecoDescricao?: string = '';
    bairro?: string = '';
	rua?: string = '';
	numero?: string = '';
	bloco?: string = '';
	apartamento?: string = '';
	complemento?: string = '';
    observacao?: string = '';
    idCliente?: number;
    telefone: string = '';
    nome: string = '';
    itensPedido: PedidoItemDTO[] = [];
    pedidoRelatorio: string = '';
}
export class PedidoItemDTO {
    id : number = 0;
    idPedido : number = 0;
    descricao : string = '';
    tipo: 'Pizza' | 'Bebida' = 'Pizza';
    valor: number = 0;
    tamanho: string = '';
    quantidade : number = 1;
    nome: string = '';
    pedidoItemPizza?: string;
}
export class PedidoItemPizzaDTO {
    pizzaSabores : PizzaSaborDTO[] = [];
}





